cli/azd/pkg/exec/runresult.go

```diff
@@ -1,15 +1,18 @@
 package exec
 
-import "fmt"
+import (
+	"fmt"
+	"os/exec"
+)
 
+// RunResult is the result of running a command.
 type RunResult struct {
+	// The exit code of the command.
 	ExitCode int
```

Comment:
I think we should remove this and move it to `ExitError`

Reply:
This caused a bug [here](https://github.com/Azure/azure-dev/pull/2186/files#r1192588270) because `err` wasn't checked.

Also, to write a correct program that handles exit codes, one actually has to write:

```golang
res, err := cmd.Run("go", "--help")
if err != nil {
   if res.ExitCode == 0 {
     // actually failed due to some other err unrelated to exit codes
   } else if res.ExitCode == 5 {
     // handle exit code 5
   } else {
    // handle non-zero exit code
   }
}
```

Reply:
We should revisit this later. Not a huge issue since `ExitCode` is by default -1 when the program doesn't run successfully.